src/js/scheduler.ts

```diff
@@ -79,7 +79,12 @@ function scheduleCallbackImmediate(callback: () => void) {
     typeof MessageChannel === "function"
   ) {
     const channel = new MessageChannel();
-    channel.port1.onmessage = () => callback();
+    channel.port1.onmessage = () => {
+      callback();
```

Comment:
unrequested nit: the original idea was based on `try/finally` which seems more appropriate https://github.com/pyodide/pyodide/issues/5597#issuecomment-2814504475

regardless, I would move this callback after the cleanup to be sure such cleanup happens regardless of the outcome of `callback()` also because if `callback()` somehow involves another scheduling the cleanup after will break that scheduling.

Reply:
Thanks for the comment! I am also +1 for your opinions.